extensions/arc/src/test/ui/dialogs/connectControllerDialog.test.ts

```diff
@@ -38,34 +38,49 @@ describe('ConnectControllerDialog', function (): void {
 
 	it('validate replaces http with https', async function (): Promise<void> {
 		await validateConnectControllerDialog(
-			{ url: 'http://127.0.0.1:30081', username: 'sa', rememberPassword: true, resources: [] },
+			{ url: 'http://127.0.0.1:30081', name: 'my-arc', username: 'sa', rememberPassword: true, resources: [] },
 			'https://127.0.0.1:30081');
 	});
 
 	it('validate appends https if missing', async function (): Promise<void> {
-		await validateConnectControllerDialog({ url: '127.0.0.1:30080', username: 'sa', rememberPassword: true, resources: [] },
+		await validateConnectControllerDialog({ url: '127.0.0.1:30080', name: 'my-arc', username: 'sa', rememberPassword: true, resources: [] },
 			'https://127.0.0.1:30080');
 	});
 
 	it('validate appends default port if missing', async function (): Promise<void> {
-		await validateConnectControllerDialog({ url: 'https://127.0.0.1', username: 'sa', rememberPassword: true, resources: [] },
+		await validateConnectControllerDialog({ url: 'https://127.0.0.1', name: 'my-arc', username: 'sa', rememberPassword: true, resources: [] },
 			'https://127.0.0.1:30080');
 	});
 
 	it('validate appends both port and https if missing', async function (): Promise<void> {
-		await validateConnectControllerDialog({ url: '127.0.0.1', username: 'sa', rememberPassword: true, resources: [] },
+		await validateConnectControllerDialog({ url: '127.0.0.1', name: 'my-arc', username: 'sa', rememberPassword: true, resources: [] },
 			'https://127.0.0.1:30080');
 	});
+
+	for (const name of ['', undefined]) {
+		it(`validate display name gets set to arc instance name for user chosen name of:${name}`, async function (): Promise<void> {
+			await validateConnectControllerDialog(
+				{ url: 'http://127.0.0.1:30081', name: name!, username: 'sa', rememberPassword: true, resources: [] },
+				'https://127.0.0.1:30081');
+		});
+	}
+
 });
 
 async function validateConnectControllerDialog(info: ControllerInfo, expectedUrl: string): Promise<void> {
+	const arcInstanceName = 'arc-instance';
+	const expectedControllerInfoName = info.name || arcInstanceName;
 	// For first set of tests just stub out refresh calls - we'll test that separately
-	sinon.stub(ControllerModel.prototype, 'refresh').returns(Promise.resolve());
 	const connectControllerDialog = new ConnectToControllerDialog(undefined!);
+	sinon.stub(ControllerModel.prototype, 'refresh').returns(Promise.resolve());
+	sinon.stub(ControllerModel.prototype, 'controllerRegistration').get(function getterFunction() {
```

Comment:
You don't need to name these functions - that's just adding unnecessary text

Reply:
done. changed to an arrow function